com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/liquibase/CheckPlatformType.java

```diff
@@ -0,0 +1,23 @@
+package com.sap.cloud.lm.sl.cf.core.liquibase;
+
+import java.text.MessageFormat;
+
+import com.sap.cloud.lm.sl.cf.core.message.Messages;
+import com.sap.cloud.lm.sl.cf.core.util.Configuration;
+
+import liquibase.database.Database;
+import liquibase.exception.CustomPreconditionErrorException;
+import liquibase.exception.CustomPreconditionFailedException;
+import liquibase.precondition.CustomPrecondition;
+
+public class CheckPlatformType implements CustomPrecondition {
+
+    @Override
+    public void check(Database database) throws CustomPreconditionFailedException, CustomPreconditionErrorException {
+        
+        String platformName = Configuration.getInstance().getPlatformType().name();
+        if(!"CF".equals(platformName)){
+            throw new CustomPreconditionFailedException(MessageFormat.format(Messages.LIQUIBASE_CF_CHECK_PLATFORM_TYPE, platformName));

```

Comment:
Should we even throw an exception here? After what we talked about in the morning this class should not even be necessary, right?

Reply:
Yes this is the way liquibase knows when the check has not pass.